web/docs/guides/websockets.md

```diff
@@ -75,55 +56,95 @@ export const webSocketFn: WebSocketDefinition<
+  chatMessage: (msg: { id: string, username: string, text: string }) => void;
+}
+
+interface ClientToServerEvents {
+  chatMessage: (msg: string) => void;
+}
+
+interface InterServerEvents {}
+
+// Data that is attached to the socket.
+// NOTE: Wasp automatically injects the JWT into the connection,
+// and if present/valid, the server adds a user to the socket.
+interface SocketData extends WaspSocketData {}
 ```
 
-## Add client support
+## Using the WebSocket on the client
 Client access to WebSockets is provided by the `useSocket` hook. It returns:
 - `socket: Socket` for sending and receiving events.
 - `isConnected: boolean` for showing a display of the Socket.IO connection status.
-  - Note: Wasp automatically connects and establishes a WebSocket connection from the client to the server by default, so you do not need to explicitly `socket.connect()` or `socket.disconnect()`. If you set `autoConnect: false` in your Wasp file, then you should call these as needed.
+  - Note: Wasp automatically connects and establishes a WebSocket connection from the client to the server by default, so you do not need to explicitly `socket.connect()` or `socket.disconnect()`. 
+  - If you set `autoConnect: false` in your Wasp file, then you should call these as needed.
 
-Additionally, there is a `useSocketListener: (socket, event, callback) => void` hook which is used for registering event handlers. It takes care of unregistering on unmount.
+Additionally, there is a `useSocketListener: (event, callback) => void` hook which is used for registering event handlers. It takes care of unregistering on unmount.
 
-All components using `useSocket` share the same underlying `socket`. Lastly, if a user is logged in, you will have a `socket.data.user` on the server.
+All components using `useSocket` share the same underlying `socket`.
 
-```ts title=src/client/Chat.tsx
-import React, { useEffect, useRef, useState } from 'react'
+```tsx title=src/client/ChatPage.tsx
+import React, { useState } from 'react'
 import { useSocket, useSocketListener } from '@wasp/webSocket'
-import { ClientToServerEvents, ServerToClientEvents } from '../shared/webSocket'
 
-export const ProfilePage = () => {
-  const [messages, setMessages] = useState<{ id: string, username: string, text: string }[]>([]);
-  const { socket, isConnected } = useSocket<ServerToClientEvents, ClientToServerEvents>()
-  const inputRef = useRef<HTMLInputElement>(null)
+export const ChatPage = () => {
+  const [messageText, setMessageText] = useState('')
+  const [messages, setMessages] = useState<
+    { id: string; username: string; text: string }[]
+  >([])
+  // The "socket" instance is typed with the types you defined on the server.
+  const { socket, isConnected } = useSocket()
 
-  useSocketListener(socket, 'chatMessage', logMessage)
+  // This is a type-safe event handler: "chatMessage" event and its payload type
+  // are defined on the server.
+  useSocketListener('chatMessage', logMessage)
```

Comment:
Nice, it is all already typed, awesome.

In the future I could imagine us wanting to define different sockets, that work with different sets of events -> but ok, that is for the future. Might be worth creating an issue for it, if we have good idea what to write. If you have some ideas here, pls open an issue, if not nvm.